design/ironic_authentication.md

```diff
@@ -0,0 +1,329 @@
+managed by cert-manager. BMO would use a custom HTTP client in
+[gophercloud](https://github.com/gophercloud/gophercloud/blob/master/docs/FAQ.md#implementing-default-logging-and-re-authentication-attempts)
+to accept that certificate. The certificate could either be self-signed or
+created using a CA given by the user. In addition, if using mTLS, a second
+certificate would be created for the BMO client from the same CA and validated
+by Apache.
+
+Considering that Ironic and Inspector are deployed in the same pod, using host
+networking, they can communicate over the loopback interface, using http or
+https. IPA can communicate with ironic and inspector over https (using
+ipa-insecure parameter to support self-signed certificates). However, it cannot
+authenticate itself using certificates if they are not built in the image. This
+process being too complex for the benefit (IPA already uses a token mechanism
+for authentication), the endpoints used by IPA will not require a client
+certificate (/v1/lookup and /v1/heartbeat for ironic, and /v1/continue for
+inspector). They will however be served with HTTPS. In addition, / and /v1/ will
+not require authentication.
+
+### Implementation Details/Notes/Constraints
+
+#### HTTPS Setup
+
+In order to simplify the HTTPS setup, when deployed in a Kubernetes cluster,
+cert-manager would be used to manage the certificates and CA. A first
+self-signed issuer would be created to generate a CA certificate. This CA
+certificate would be used to create a second issuer. This second issuer would
+be used to generate the server certificates (ironic and inspector), and the
+client certificates (BMO and CLI).
+
+A naming convention would allow the use of the same CA for the certificate for
+BMO (deployed as part of CAPM3 for example, or on its own), and for the
+certificate for the Ironic pod containers.
+
+In order to share the CA, there is two alternatives :
+
+* BMO and ironic are deployed in the same namespace, they can then share the
+  same issuer. However, this adds a requirement for the deployments to share
+  the same namespace
+* BMO and Ironic are deployed in different namespaces. In this case they must
+  share a clusterIssuer. This would work across namespaces, however, a
+  certificate could be requested from any other namespace, as long as the user
+  creating it has the proper RBAC rights to create a Certificate object. This
+  means that it would be very straightforward to get a certificate and
+  authenticate as an ironic client. This solution should not be considered.
+
+In order to use a properly secured setup with HTTPS, BMO and ironic should be
+deployed in the same namespace, using the same CA, created by cert-manager.
+
+It should be ensured that a rotation of the certificates or of the CA would be
+effective immediately for all pieces of the setup.
+
+#### Mutual TLS
+
+This option is more secure. The CA and the certificates would be managed by
```

Comment:
Now that this is the main proposal, this needs to say what it is more secure than :)

Reply:
sure, thanks